docs/dev_guides/changing_splink/dependency_management.md

```diff
@@ -0,0 +1,93 @@
+## Managing Splink's Dependencies
+
+This page highlights the importance of package versioning and proposes that we use a "sunsetting" strategy for updating our support python and dependency versions as they reach end-of-life.
+
+Additionally, it lays out some rough guidelines for us to follow when addresses future package conflicts and issues arises from antiquated dependency versions.
+
+<hr>
+
+## Package Versioning Policy
+
+Monitoring package versioning within Splink is important. It ensures that the project can be used by as wide a group of individuals as possible, without wreaking havoc on our issues log.
+
+Below is a rough summary of versioning and some complimentary guidelines detailing how we should look to deal with dependency management going forward.
+
+### Benefits to Effective Versioning
+
+Effective versioning is crucial for ensuring Splink's compatibility across diverse devices and seamless integration with various Python versions and cloud tools. Key advantages include:
+
+* Faster dependency resolution with `poetry lock`.
+* Reduces dependency conflicts across systems.
+* And, as Robin put it, *"Make things 'just work' more easily in the 99% of cases where versions do not cause an issue"*.
+
+### Versioning Guidance
+
+**Establish Minimum Supported Versions**
+
+* **Align with Python Versions**: Select the minimum required versions for dependencies based on the earliest version of Python we plan to support. This approach is aligned with our policy on [**Sunsetting End-of-Life Python Versions**](#sunsetting-end-of-life-python-versions), ensuring Splink remains compatible across a broad spectrum of environments.
+* **Document Reasons**:  Where appropriate, clearly document why specific versions are chosen as minimums, including any critical features or bug fixes that dictate these choices. We should look to do this in pull requests implementing the change and/or as comments in [`pyproject.toml`](https://github.com/moj-analytical-services/splink/blob/9499e4ee93e6157fcc6f228b60592a7cf97bb6a0/pyproject.toml#L143). Doing so allows us to easily track versioning decisions.
+
+**Prefer Open Version Constraints**
+
+* **Use Open Upper Bounds**: Wherever feasible, avoid setting an upper version limit for a dependency. This reduces compatability conflicts with external packages and allows the user to decide their versioning strategy at the application level.
+* **Monitor Compatibility**: Actively monitor the development of our core dependencies to anticipate significant updates (such as new major versions) that might necessitate code changes. Within Splink, this is particularly relevant for both Sqlglot and :simple-duckdb:, that (semi)frequently release new, breaking changes.
```

Comment:
Change to SQLGlot (as per their docs)

Also, include the name "DuckDB" after the emoji